torchtitan/config_manager.py

```diff
@@ -347,6 +347,18 @@ def __init__(self):
                 here: https://github.com/pytorch-labs/float8_experimental
             """,
         )
+        self.parser.add_argument(
```

Comment:
As discussed offline, let's refactor fp8 configs, e.g. have a dedicated field for enabling fp8 or not.

Reply:
renamed `fp8_linear` to `enable_fp8_linear`

Reply:
I think one thing to note is that right now this is a boolean which will swap to the default float8 recipe
Dynamic scaling x Tensor wise ScalingGranularity x all tensors involved in the matmul [ input, weight, grad]

I think we should brainstorm on an elegant solutions for users to express their desired config here

Reply:
good question. evetually we might have to expose args/kwargs from `swap_linear_with_float8_linear` for flexibility